### DISCUSSION TITLE
How to refresh table heading?

### DISCUSSION THREAD

**Discussion by azzarip** [filamentphp/filament#13284] [Help]
### Package

Panel builder

### Package Version

v3.2.40

### How can we help you?

Hi everyone, 

I have a Relationship Manager of belongs to many relationship. 
I have an attach action that works perfectly, but I would like to show on the heading of the relationship table the count of items. 

I set the heading in the table method as shown below, but when I attach a new relationship, it does not update. The only way I found so far is to reload the page (see below).

```php
    public function table(Table $table): Table
    {
        return $table
           ->heading('Count: ' . $this->ownerRecord->contacts->count())
           ->columns(...)
           ->headerActions([
                Tables\Actions\AttachAction::make()
                ->color('info')
                ->preloadRecordSelect()
                ->after(function (){
                    $this->js('window.location.reload()');
                }),
            ])
```
Is there any other way to do it?
I tried with `$this->resetPage()` or `$this->resetTable()`, both unsuccessful.

If I detach a record, without any additional command, it refreshes the heading, and the count goes down.

**Comment by leandrocfe** [comment]
`$this->getTable()->heading('Count: ' . $this->ownerRecord->contacts->count())`
?

**Reply by azzarip** [reply]
No, does not work. 

Fun fact: if I detach a record the heading gets refreshed. And the count goes down.

**Comment by azzarip** [comment]
Find a fix instead of using
`$this->ownerRecord->contacts->count()` 
I query it directly on the database and it updates.

`DB::table('contact_tag')->where('tag_id', $this->ownerRecord->id)->count();` 

so the after becomes:
```php
->after(function (){
    $this->getTable()->heading($this->getContactCount());
}),
``` 

and 

```php 

    protected function getContactCount(): string
    {
        return 'Contacts: ' . DB::table('contact_tag')->where('tag_id', $this->ownerRecord->id)->count();
    }
```

**Comment by talkinnl** [comment]
Old topic, but I think it's better to wrap the count bit in a closure, so it gets evaluated at a later point in time.

(Sounds a bit like React props issue ;) )
